### PR TITLE
Properly redirect healthcheck endpoint if using BaseUrl

### PR DESCRIPTION
**Pull request by crobibero**
Now `/health` can be checked automagically even when setting a baseUrl

### CODE REVIEW

**Comment by BaronGreenback** [comment]
Cheeky request - but how about also adding a base redirect to the wizard screen if the config setup isn't complete?

**Review by BaronGreenback — APPROVED** [review]

**Comment by crobibero** [comment]
> 
> 
> Cheeky request - but how about also adding a base redirect to the wizard screen if the config setup isn't complete?

Wizard screen is a web issue, server knows nothing of it

**Comment by BaronGreenback** [comment]
thought of using a redirect to wizard screen if <IsStartupWizardCompleted>true</IsStartupWizardCompleted> is false.

Did a reinstall this week, and couldn't get to screen as i couldn't remember the url. Opening bookmark, just caused an error and hung.

**Review by Ullmie02 — APPROVED** [review]

File: Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs [hunk 1]
```diff
@@ -45,16 +45,27 @@ public async Task Invoke(HttpContext httpContext, IServerConfigurationManager se
...
-                || string.Equals(localPath, "/", StringComparison.OrdinalIgnoreCase)
-                || string.IsNullOrEmpty(localPath)
-                || !localPath.StartsWith(baseUrlPrefix, StringComparison.OrdinalIgnoreCase))
+            if (!string.IsNullOrEmpty(baseUrlPrefix))
             {
-                // Always redirect back to the default path if the base prefix is invalid or missing
-                _logger.LogDebug("Normalizing an URL at {LocalPath}", localPath);
-                httpContext.Response.Redirect(baseUrlPrefix + "/" + _configuration[ConfigurationExtensions.DefaultRedirectKey]);
-                return;
+                if (!localPath.StartsWith(baseUrlPrefix, StringComparison.OrdinalIgnoreCase)
```
Comment (cvium) [inline Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs:50]: You're performing this check twice

File: Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs [hunk 2]
```diff
@@ -45,16 +45,29 @@ public async Task Invoke(HttpContext httpContext, IServerConfigurationManager se
...
+                    _logger.LogDebug("Redirecting /health check");
+                    httpContext.Response.Redirect(baseUrlPrefix + "/health");
+                    return;
+                }
+
+                if (string.Equals(localPath, baseUrlPrefix + "/", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(localPath, baseUrlPrefix, StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(localPath, "/", StringComparison.OrdinalIgnoreCase)
+                    || string.IsNullOrEmpty(localPath)
+                    || !startsWithBaseUrl)
```
Comment (cvium) [inline Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs:64]: The order and expressions could be improved wrt. fast path, allocations etc. I'm also wondering why localPath == '/' is checked?
```suggestion
                if (!startsWithBaseUrl
                    || string.IsNullOrEmpty(localPath)
                    || localPath.Length == baseUrlPrefix.Length
                    || (localPath.Length == baseUrlPrefix.Length + 1 && localPath.EndsWith('/')))
```

Comment (crobibero) [reply Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs]: I've simplified it further and fixed an issue where requesting `/health/` wouldn't properly redirect. Everything still redirects properly for me with this change

**Review by cvium — APPROVED** [review]

**Comment by LinFor** [comment]
Seems that default redirect (/ to /web/index.html) is broken now if baseUrlPrefix empty.

**Pull request merged**